pyface/tests/test_widget.py

```diff
@@ -315,3 +319,16 @@ def test_enabled(self):
                 self.widget.enabled = False
 
         self.assertFalse(self.widget.control.isEnabled())
+
+    @unittest.skipIf(
+        is_linux,
+        "Linux keyboard focus is False unless window is active",
+    )
+    def test_focus(self):
+        with self.event_loop():
+            self.widget.create()
+
+        with self.event_loop():
+            self.widget.focus()
```

Comment:
```suggestion
        with self.event_loop():
            self.widget.create()

        self.assertFalse(self.widget.has_focus())

        with self.event_loop():
            self.widget.focus()
```

Reply:
I was cautious about this, because that may depend on OS behaviour (the OS may automatically put keyboard focus in the first widget in the active window), but I'm willing to give it a shot and see if it works.